zkvm/src/ops.rs

```diff
@@ -277,3 +279,213 @@ impl Instruction {
+
+    pub fn range(&mut self, bit_width: u8) -> &mut Program {
+        self.0.push(Instruction::Range(bit_width));
+        self
+    }
+
+    pub fn and(&mut self) -> &mut Program {
+        self.0.push(Instruction::And);
+        self
+    }
+
+    pub fn or(&mut self) -> &mut Program {
+        self.0.push(Instruction::Or);
+        self
+    }
+
+    pub fn verify(&mut self) -> &mut Program {
+        self.0.push(Instruction::Verify);
+        self
+    }
+
+    pub fn blind(&mut self) -> &mut Program {
+        self.0.push(Instruction::Blind);
+        self
+    }
+
+    pub fn reblind(&mut self) -> &mut Program {
+        self.0.push(Instruction::Reblind);
+        self
+    }
+
+    pub fn unblind(&mut self) -> &mut Program {
+        self.0.push(Instruction::Unblind);
+        self
+    }
+
+    pub fn issue(&mut self) -> &mut Program {
+        self.0.push(Instruction::Issue);
+        self
+    }
+
+    pub fn borrow(&mut self) -> &mut Program {
+        self.0.push(Instruction::Borrow);
+        self
+    }
+
+    pub fn retire(&mut self) -> &mut Program {
+        self.0.push(Instruction::Retire);
+        self
+    }
+
+    pub fn qty(&mut self) -> &mut Program {
+        self.0.push(Instruction::Qty);
+        self
+    }
+
+    pub fn flavor(&mut self) -> &mut Program {
+        self.0.push(Instruction::Flavor);
+        self
+    }
+
+    pub fn cloak(&mut self, m: usize, n: usize) -> &mut Program {
+        self.0.push(Instruction::Cloak(m, n));
+        self
+    }
+
+    pub fn import(&mut self) -> &mut Program {
+        self.0.push(Instruction::Import);
+        self
+    }
+
+    pub fn export(&mut self) -> &mut Program {
+        self.0.push(Instruction::Export);
+        self
+    }
+
+    pub fn input(&mut self) -> &mut Program {
+        self.0.push(Instruction::Input);
+        self
+    }
+
+    pub fn output(&mut self, payload_count: usize) -> &mut Program {
+        self.0.push(Instruction::Output(payload_count));
+        self
+    }
+
+    pub fn contract(&mut self, payload_count: usize) -> &mut Program {
+        self.0.push(Instruction::Contract(payload_count));
+        self
+    }
+
+    pub fn nonce(&mut self) -> &mut Program {
+        self.0.push(Instruction::Nonce);
+        self
+    }
+
+    pub fn log(&mut self) -> &mut Program {
+        self.0.push(Instruction::Log);
+        self
+    }
+
+    pub fn sign_tx(&mut self) -> &mut Program {
+        self.0.push(Instruction::Signtx);
+        self
+    }
+
+    pub fn call(&mut self) -> &mut Program {
+        self.0.push(Instruction::Call);
+        self
+    }
+
+    pub fn left(&mut self) -> &mut Program {
+        self.0.push(Instruction::Left);
+        self
+    }
+
+    pub fn right(&mut self) -> &mut Program {
+        self.0.push(Instruction::Right);
+        self
+    }
+
+    pub fn delegate(&mut self) -> &mut Program {
+        self.0.push(Instruction::Delegate);
+        self
+    }
+
+    pub fn ext(&mut self, n: u8) -> &mut Program {
```

Comment:
You are not supposed to use extension opcodes until they are actually defined. So i'd remove this from the API.